src/op_utils.rs

```diff
@@ -66,15 +66,14 @@ fn test_arg_count() {
     assert_eq!(arg_count(&count_3_args, 4), 3);
 }
 
-pub fn int_atom<'a>(args: &'a Node, op_name: &str) -> Result<&'a [u8], EvalErr> {
+pub fn int_atom<'a>(args: Node<'a>, op_name: &str) -> Result<&'a [u8], EvalErr> {
     match args.atom() {
         Some(a) => Ok(a),
         _ => args.err(&format!("{op_name} requires int args")),
     }
 }
 
-// rename to atom()
-pub fn atom<'a>(args: &'a Node, op_name: &str) -> Result<&'a [u8], EvalErr> {
+pub fn atom<'a>(args: Node<'a>, op_name: &str) -> Result<&'a [u8], EvalErr> {
```

Comment:
these are the two main changes, the other changes fall out of these.
Note that the returned buffer depends on the allocator's lifetime (which is the lifetime parameter to `Node` itself) not on the reference to the node itself (which is what we used to do).

The same goes for `int_atom()` above.

These changes allow passing in temporary `Node` objects to these functions, and make code using them more compact, without unnecessary new variables.